### PR TITLE
Fixed code for KDeep reproduction

### PR DESCRIPTION
**Pull request by hassanmohsin**
The scripts required to reproduce the KDeep results are fixed. However, the result is not the same. Check the plots and Bench.hdh5

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,8 +1,132 @@
-# IDE-Files
-.idea
+# Byte-compiled / optimized / DLL files
```
Comment (BioFreak95) [inline .gitignore:1]: I do not get, why you delete my ignore-Stuff :D I think this is very personal? You ignore much stuff, which is not commonly part of the package. Maybe I missunderstood something?

File: scripts/CNN/CNN_Analysis.py [hunk 2]
```diff
@@ -38,8 +38,8 @@
 plt.xlabel('Target')
 plt.ylabel('Output of Network')
 plt.title('Output of Pytorch-trained Network for Benchmark-Set')
```
Comment (BioFreak95) [inline scripts/CNN/CNN_Analysis.py:40]: This xlim and ylim are just proposals. Usually it make sense, because the data only have points between 2 and 12.

File: scripts/CNN/CNN_Preprocessing.py [hunk 3]
```diff
@@ -1,9 +1,8 @@
 import sys
-
-sys.path.append('/home/max/Dokumente/Masterarbeit/PredBind')
```
Comment (BioFreak95) [inline scripts/CNN/CNN_Preprocessing.py:3]: Thanks for this change!

File: scripts/CNN/CNN_Preprocessing.py [hunk 4]
```diff
@@ -25,31 +24,44 @@
...
+#     Preprocessing.convertData(namespace=protNamespace, startExt='pdb', targetExt='mol2', proteinnumber=i,
+#                               path=trainPath)
 
 # Try first with pdb if this not work, change it to mol2 -> The reason is, that in pdbbind the data comes
 # from experiments and can sometimes a little bit broken -> In this case it can not be readed -> converting can help
 
+print("LOG: Creating voxelized file")
 prep.createVoxelisedFile(datapath='../../Data/train/', savepointnum=50, protNamespace='_protein.pdb',
                          ligNamespace='_ligand.mol2', namespace='../../Data/temp/train/train',
-                         altProNamespace='_protein.mol2', altLigNamespace='_ligand.pdb', startpoint=600)
```
Comment (BioFreak95) [inline scripts/CNN/CNN_Preprocessing.py:40]: Also very senseful, I added the data step by step, so I added this feature.

File: scripts/CNN/CNN_Pytorch.py [hunk 5]
```diff
@@ -1,21 +1,21 @@
 import sys
-sys.path.append('/home/max/Dokumente/Masterarbeit/PredBind')
```
Comment (BioFreak95) [inline scripts/CNN/CNN_Pytorch.py:2]: Thanks for this change

File: src/CNN/networks/Training.py [hunk 6]
```diff
@@ -144,7 +145,7 @@ def benchmark(self, n_datapoints, datapath, rotations=True, model=None, ensemble
                 for j in range(24):
                     with h5py.File(datafile, 'r') as file:
                         data = rot.rotation(data=file[str(i) + '/data'][()][0], k=j)
-                        label = -np.log10(np.exp(-(file[str(i) + '/label'][()])))
```
Comment (BioFreak95) [inline src/CNN/networks/Training.py:147]: Thanks. I had to add this, because I had to change the natural log to log10, because the old code used log2 instead of log10.

File: src/CNN/networks/Training.py [hunk 7]
```diff
@@ -210,25 +211,19 @@ def fit(self, epochs, train_path, result_datapath, kwargs=None, n_datapoints=376
...
 
-        if test_path is None:
-            test_set = OwnDataset(test, train_path, rotations=False)
-        else:
-            test_set = OwnDataset(np.arange(n_test), test_path, rotations=False)
+#         indices = np.arange(n_datapoints)
+#         train_size = int(prct_train * n_datapoints)
+#         test_size = n_datapoints - train_size
+#         train, test = torch.utils.data.random_split(indices, [train_size, test_size])
+        # Use the entire refined set as train set and core set as test/validation set
```
Comment (BioFreak95) [inline src/CNN/networks/Training.py:220]: The Test-set and Validation-set should never be the same.

File: src/CNN/networks/layers/OwnDataset.py [hunk 8]
```diff
@@ -25,7 +25,7 @@ def __getitem__(self, index):
             else:
                 data = file[str(data_idx) + '/data'][()][0]
             label = file[str(data_idx) + '/label'][()]
-            label = -np.log10(np.exp(-label))
```
Comment (BioFreak95) [inline src/CNN/networks/layers/OwnDataset.py:28]: Same as above. Old code used log2 instead of log10 so I had to recalc this. So if one uses the new code for calculating the data, your change is correct. Thanks!

File: src/CNN/tools/Preprocessing.py [hunk 9]
```diff
@@ -133,7 +133,7 @@ def calcProtVoxel(self, x, y, z, protPath, number, altProtPath):
                     )
                 except:
                     f = open("../../Data/prep_log.txt", "a")
-                    f.writelines('Protein ' + protPath + ' leads to errors! Proteinnumber: ' + str(number) + '\n')
```
Comment (BioFreak95) [inline src/CNN/tools/Preprocessing.py:136]: The number is very usefull, because you know, which starting point you should use after the fix. This avoids recalc everything.

File: src/CNN/tools/Preprocessing.py [hunk 10]
```diff
@@ -196,20 +196,24 @@ def getLabels(complex_data_path, index_path='Data/index/INDEX_refined_data.2016'
         if complexnames is None:
             complexnames = Preprocessing.getComplexDirNames(complex_data_path)
         with open(index_path) as f:
-            reader = csv.reader(f, delimiter='\t')
```
Comment (BioFreak95) [inline src/CNN/tools/Preprocessing.py:199]: depends on the data. classically this is a tab

File: src/CNN/tools/Preprocessing.py [hunk 11]
```diff
@@ -196,20 +196,24 @@ def getLabels(complex_data_path, index_path='Data/index/INDEX_refined_data.2016'
...
             index = complexes.index(complexnames[i])
-            label = Preprocessing.extractKValue(datas[index])
+            label = float(datas[index][3])
+            #label = Preprocessing.extractKValue(datas[index])
             labels.append(label)
-        for i in range(len(labels)):
-            labels[i] = -np.log10(labels[i])
+        #for i in range(len(labels)):
+        #    labels[i] = -np.log10(labels[i])
         return labels
```
Comment (BioFreak95) [inline src/CNN/tools/Preprocessing.py:218]: For this change, there is already a second method, which reads the value and do not calc it. Is faster but not that precice

**Review by maximilian-salomon — COMMENTED** [review]
Thanks for the changes. Have make some comments, maybe you can say something to this with questions. Results-Files should not be part of the upload. Maybe you can delete it.

In generel I have the questions, what you are changing. You use pybel/babel to convert mol2 to pdb, but with a script and not a method. I am doing the same with the convertData method. I dont really get your changes in the converting. Maybe you can explain it?

Nevertheless, it was possible for me, to get the same results as in the KDeep-paper in respect to the MSE of the Deep-Learning algorithm. So my network-performance was the same (more or less) as in the paper. So what do you mean with other results? Do you mean in respect to the data or to the network performance?

**Comment by BioFreak95** [comment]
Nevermind. I saw the plots now. This looks special. I do not know, what happend. It seems, that you are overfitting. I had much less overfitting and better results. I can upload a figure.

File: .gitignore (hunk 1 above)
Comment (hassanmohsin) [reply .gitignore:1]: Nevermind. I just used the github template

**Comment by BioFreak95** [comment]
Reproduction_Full_Output
TrainHist_Rot

**Comment by BioFreak95** [comment]
Reproduction_Full_Output

File: src/CNN/networks/Training.py (hunk 7 above)
Comment (hassanmohsin) [reply src/CNN/networks/Training.py:220]: I am afraid the Kdeep paper used the test set as validation set!!

**Comment by BioFreak95** [comment]
Reproduction_Full_Output

**Comment by BioFreak95** [comment]
The History is the result of a 85/15 train-val split. The other is trained on the full train data and tested on the full test-data. The picture is the output of the test-run.

**Comment by hassanmohsin** [comment]
I also trained for 100 epochs. Not sure why I got a very different result! Any thoughts?

**Comment by hassanmohsin** [comment]
`convertData` method was throwing segmentation fault. So, the conversion process was taken outside of the script using openbabel directly instead of pybel.

**Comment by hassanmohsin** [comment]
I used all 3767 samples in the refined set as the training set and 290 samples in the core set as the validation set (as in kdeep). My plot also reflects that.

**Comment by BioFreak95** [comment]
Deleted the figures, because they are from my thesis, which is not evulated yet.
Give me some time to think about it. I answer this eve.
